test/fail_compilation/README.md

```diff
@@ -0,0 +1,36 @@
+# README For fail_compilation tests
+
+Each file with a .d extension is run and is expected to fail to compile.
+The diagnostic emitted must match the diagnostic in the test file,
+and if it does, the test passes.
+
+## Purpose
+
+The point of these files is to test that the compiler produces a correct
+diagnostic for each error message in the compiler's implementation.
+The ideal is to achieve 100% coverage of the compiler source code.
+
+A further aim is that when the compiler does fail these tests, the test case
+should be crafted to make debugging the compiler as straightforward as practical.
+
+## Guidelines
+
+* Do not import any phobos or druntime files.
+* If a test is added as a result of a bugizilla issue, include
+a clickable URL to the corresponding bugzilla issue.
+* It takes a long time to run these test. The run time corresponds with
+the number of test files, not the size of the test file. Therefore, it makes
+sense to group many tests into one file instead of multiple files.
```

Comment:
Perhaps also mention that, by default, DMD will only output up to 20 error messages, and that can be changed via `-verrors=<num>`

Reply:
Nope. That flag is enabled by default.